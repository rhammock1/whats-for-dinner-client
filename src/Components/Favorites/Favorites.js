/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { Link } from 'react-router-dom';
import apiService from '../../services/api-service';
import './Favorites.css';

const FavoriteRecipe = (props) => {
  const { recipe } = props || {};

  return (
    <div className="detail">
      <p>
        <Link className="detail-link" to={`/recipes/${recipe.id}`}>
          {recipe.title}
          {' '}
        </Link>
        <span id="style">recipe</span>
      </p>
    </div>
  );
};

const FavoriteRestaurant = (props) => {
  const { restaurant } = props || {};
  return (
    <div className="detail">
      <p>
        <Link className="detail-link" to={`/restaurants/${restaurant.id}`}>
          {restaurant.title}
          {' '}
        </Link>
        <span id="style">{restaurant.style}</span>
      </p>
    </div>
  );
};

class Favorites extends React.Component {
  static defaultProps = {
    match: {
      params: {
        userId: 0,
      },
    },
  }

  state = {
    error: null,
    favorites: [],
    recipes: [],
    restaurants: [],
  }

  componentDidMount() {
    const { userId } = this.props.match.params;
    apiService.getUsersThings(userId, 'favorites')
      .then((favorites) => this.setState({ favorites }))
      .catch((error) => this.setState({ error }));
    apiService.getUsersThings(userId, 'recipes')
      .then((recipes) => this.setState({ recipes }))
      .catch((error) => this.setState({ error }));
    apiService.getUsersThings(userId, 'restaurants')
      .then((restaurants) => this.setState({ restaurants }))
      .catch((error) => this.setState({ error }));
  }

  handleBuildFavorites = (favorite) => {
    const whatItIS = `${favorite.what_it_is}s`;
    const { restaurants, recipes } = this.state;
    let favoriteThing;
    if (whatItIS === 'restaurants') {
      favoriteThing = restaurants.filter((restaurant) => restaurant.id === favorite.item_id);

      return (
        <FavoriteRestaurant key={favorite.id} restaurant={favoriteThing} />
      );
    }
    if (whatItIS === 'recipes') {
      favoriteThing = recipes.filter((recipe) => recipe.id === parseFloat(favorite.item_id));

      return (
        <FavoriteRecipe key={favorite.id} recipe={favoriteThing} />
      );
    }
    return favoriteThing;
  }

  render() {
    const { favorites, error } = this.state;

    return (
      <section className="favorites">
        <h2>My Favorites</h2>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="favorites-container">
          {favorites.map((favorite) => this.handleBuildFavorites(favorite))}
        </div>
      </section>

    );
  }
}

export default Favorites;
