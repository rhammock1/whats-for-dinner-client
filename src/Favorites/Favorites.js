import React from 'react';
import { Link } from 'react-router-dom';
import apiService from '../services/api-service';


const FavoriteRecipe = function(props) {
  const recipe = props.recipe[0] || {}
  
  return (
     <div>
        <Link to={`/recipes/${recipe.id}`}><p>{recipe.title}</p></Link>
      </div>
    )
}

const FavoriteRestaurant = function(props) {
  
  const restaurant = props.restaurant[0] || {}
  return (
    <div>
      <Link to={`/restaurants/${restaurant.id}`}><p>{restaurant.title} <span id='style'>{restaurant.style}</span></p></Link>
    </div>
  )
  
}



class Favorites extends React.Component {

  state = {
    favorites: [],
    recipes: [],
    restaurants: [],
  }


  componentDidMount() {
    const {userId} = this.props.match.params
    apiService.getUsersThings(userId, 'favorites')
      .then(favorites => this.setState({favorites}))
      .catch(error => console.error(error))
    apiService.getUsersThings(userId, 'recipes')
      .then(recipes => this.setState({recipes}))
      .catch(error => console.error(error))
    apiService.getUsersThings(userId, 'restaurants')
      .then(restaurants => this.setState({restaurants}))
      .catch(error => console.error(error))
}

  handleBuildFavorites = (favorite) => {
    const whatItIS = `${favorite.what_it_is}s`
    let favoriteThing;
    if(whatItIS === 'restaurants') {
      favoriteThing = this.state.restaurants.filter(restaurant => restaurant.id === favorite.item_id)
      
      return (
        <FavoriteRestaurant key={favoriteThing.id} restaurant={favoriteThing} />
      )
    }
    if(whatItIS === 'recipes') {
      favoriteThing = this.state.recipes.filter(recipe => recipe.id === parseFloat(favorite.item_id))
      
      return (
        <FavoriteRecipe key={favoriteThing.id}recipe={favoriteThing} />
      )
    }
  }
  render() {
    const favorites = this.state.favorites;

    return (
      <section className='favorites'>
        <h2>My Favorites</h2>
        <div className='favorites-container'>
          {favorites.map(favorite => {
            return this.handleBuildFavorites(favorite)
            
          })}
        </div>
      </section>

      )
  }
}

export default Favorites;