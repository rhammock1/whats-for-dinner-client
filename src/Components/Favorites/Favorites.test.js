  
import React from 'react';
import ReactDOM from 'react-dom';
import Favorites from './Favorites';
import { BrowserRouter } from 'react-router-dom';

describe('Favorites component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Favorites />
    </BrowserRouter>, div)
  })
})