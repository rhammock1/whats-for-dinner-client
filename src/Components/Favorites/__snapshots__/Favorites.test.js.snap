// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component renders the complete form 1`] = `
<section
  className="favorites"
>
  <h2>
    My Favorites
  </h2>
  <div
    className="favorites-container"
  />
</section>
`;
